### ISSUE TITLE
Configuration should not use env var but `config-kourier`

### ISSUE THREAD

**Issue by nak3** [knative-extensions/net-kourier#877]
Currently some configurations especially external authorization use env variables as https://github.com/knative-sandbox/net-kourier#external-authorization-configuration instead of config-kourier configmap.

I think that there is no big reason for using env variable but we just used it because we haven't introduced `config-kourier` at that time.

Now, since new configurations are added in `config-kourier`, the env variables should be migrated into the configmap.

**Comment by ReToCode** [comment]
The same would probably apply for `CERTS_SECRET_NAMESPACE` and `CERTS_SECRET_NAME` used in `caches.go`

**Comment by harshil1973** [comment]
I would like to contribute in this but I am an absolute beginner if anyone can help me by providing steps it would be nice

**Comment by nak3** [comment]
Hmm... I think it is not a "good-first-issue" issue actually it needs many steps for the change and one big concern is that some configurations might need to be loaded before ConfigMap informer's reconcile loop.
How do you think @ReToCode

**Comment by ReToCode** [comment]
Yeah, I think it could be more complicated than initially though. I haven't checked if any of the configs actually is affected by what you described. But it is probably better to start with a simpler issue @harshil1973.

/remove-good-first-issue

**Comment by ystkfujii** [comment]
I made it possible to configure External Authorization and TLS certificate settings via ConfigMap. 
Is there anything else that should be addressed in this issue?

- https://github.com/knative-extensions/net-kourier/pull/1334
- https://github.com/knative-extensions/net-kourier/pull/1352
